### PR TITLE
@types/rollup: update bundle and write options to v0.50

### PR DESCRIPTION
**Pull request by goser**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/rollup/rollup/blob/master/src/utils/deprecateOptions.js
- [x] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/rollup/index.d.ts [hunk 1]
```diff
@@ -24,12 +24,12 @@ export interface Warning {
 export interface BundleOptions {
 	/** The format of the generated bundle. */
 	format?: Format
```
Comment (flying-sheep) [inline types/rollup/index.d.ts]: you missed this deprecation:

https://github.com/rollup/rollup/blob/b949eb08169115ff66648838cbc4833379bf9440/src/utils/deprecateOptions.js#L60

File: types/rollup/index.d.ts [hunk 2]
```diff
@@ -55,26 +55,26 @@ export interface BundleOptions {
...
 
 export interface WriteOptions extends BundleOptions {
-	/** The file to write to. If `options.sourceMap === true`, two files will be created – `dest` and `dest + '.map`. */
-	dest: string
-	/** If `true`, a separate sourcemap file will be created. If `'inline'`, the sourcemap will be appended to the resulting dest file as a data URI. */
-	sourceMap?: boolean | 'inline'
-	/** This option is unnecessary, as it defaults to the value of dest. */
-	sourceMapFile?: string
+	/** The file to write to. If `options.sourcemap === true`, two files will be created – `file` and `file + '.map`. */
+	file: string
```
Comment (flying-sheep) [inline types/rollup/index.d.ts]: this is wrong, it’s <code>options<em>.output</em>.file</code>:

https://github.com/rollup/rollup/blob/b949eb08169115ff66648838cbc4833379bf9440/src/utils/deprecateOptions.js#L50

**Review by flying-sheep — CHANGES_REQUESTED** [review]
hi, thanks for doing this! i found [this gist](https://gist.github.com/Rich-Harris/d472c50732dab03efeb37472b08a3f32) which documents the changes i commented on.

also you should also add missing options like `name`, `watch`, `amd`, `extend`, and maybe more. you can find those in the changelog, and by comparing the docs with these definitions.

File: types/rollup/index.d.ts (hunk 2 above)
Comment (goser) [reply types/rollup/index.d.ts]: As far a I understand the part "[Changes to the API](https://gist.github.com/Rich-Harris/d472c50732dab03efeb37472b08a3f32#changes-to-the-api)" and the source code the deprecateOptions script is only used for the config file. Not for the API. 

I assumed the type definition was meant to be used for the API only.

Comment (flying-sheep) [reply types/rollup/index.d.ts]: ah i see. so we have to separate the write options from the config file options.

Comment (andrewiggins) [reply types/rollup/index.d.ts]: If anyone was curious the [rollup/index.js file](https://github.com/rollup/rollup/blob/master/src/rollup/index.js#L66) contains the deprecation checks for the API.

File: types/rollup/index.d.ts [hunk 3]
```diff
@@ -24,12 +24,12 @@ export interface Warning {
 export interface BundleOptions {
 	/** The format of the generated bundle. */
 	format?: Format
-	/** What export mode to use. Defaults to auto, which guesses your intentions based on what the `entry` module exports. */
+	/** What export mode to use. Defaults to auto, which guesses your intentions based on what the `import` module exports. */
 	exports?: 'auto' | 'default' | 'named' | 'none'
 	/** An ID to use for AMD/UMD bundles. */
 	moduleId?: string
```
[the remark this answers is not in the document]
Comment (andrewiggins) [reply types/rollup/index.d.ts]: In favor of `amd: { id: string }`

**Review by andrewiggins — COMMENTED** [review]
@goser Thanks for doing this update! Looking forward to its completion.

**Comment by andrewiggins** [comment]
@flying-sheep Any other comments on this PR?

**Comment by flying-sheep** [comment]
nope

**Comment by andy-ms** [comment]
@goser It looks like there are some unaddressed comments.

**Comment by RyanCavanaugh** [comment]
@goser Please fix the failures indicated in the Travis CI log.

**Comment by andrewiggins** [comment]
@goser, your build has some linting errors, copied below for convenience. Can you fix these? If you don't have time, I'm happy to take over the changes.

Thanks for getting this started!

```
Error: /home/travis/build/DefinitelyTyped/DefinitelyTyped/types/rollup/index.d.ts
ERROR: 31:7   typedef-whitespace  expected onespace after colon in property-declaration
ERROR: 31:13  typedef-whitespace  expected onespace after colon in property-declaration
ERROR: 31:29  typedef-whitespace  expected onespace after colon in property-declaration
ERROR: 31:7   whitespace          missing whitespace
ERROR: 31:13  whitespace          missing whitespace
ERROR: 31:29  whitespace          missing whitespace
```

**Comment by minestarks** [comment]
@goser I'd love to merge this in but I can't until the tests are fixed. Could you make changes to avoid the PR getting closed due to staleness? Thanks.

**Comment by minestarks** [comment]
I'm closing this PR due to no activity.

**Pull request closed without merging**